xalia/UiDom/UiDomElement.cs

```diff
@@ -377,12 +380,10 @@ protected override UiDomValue EvaluateIdentifierCore(string id, UiDomRoot root,
                 case "assign":
                     return new UiDomMethod(this, "assign", AssignFn);
                 case "index_in_parent":
-                    if (!(Parent is null))
-                    {
-                        depends_on.Add((Parent, new IdentifierExpression("children")));
-                        return new UiDomInt(Parent.Children.IndexOf(this));
-                    }
-                    return UiDomUndefined.Instance;
+                    if (Parent is null)
+                        return UiDomUndefined.Instance;
+                    depends_on.Add((Parent, new IdentifierExpression("children")));
+                    return new UiDomInt(IndexInParent);

```

Comment:
I'd guess it's unlikely to ever happen, but further down in the getter for `IndexInParent` we have:
```
                if (Parent is null)
                    return -1;
```
Which IIUC would mean this would return `-1` instead of `UiDomUndefined.Instance` in the null parent case. Might not be a big deal, but just thought I'd point it out.

Reply:
Cool, just wanted to make sure this wasn't missed. :)